github/ci/prow-deploy/kustom/overlays/workloads-production/resources/priority-classes.yaml

```diff
@@ -16,3 +16,12 @@ value: 1000000
 preemptionPolicy: Never
 globalDefault: false
 description: "Allows gpu jobs to be scheduled with higher priority."
+---
+apiVersion: scheduling.k8s.io/v1
+kind: PriorityClass
+metadata:
+  name: sriov-place-holder
+value: 1000000
```

Comment:
So IIUC the sriov place holder pods will have higher priority than the normal pods running with default priority.

K8s docs say:
> Preemption considers Pod priority and attempts to choose a set of targets with the lowest priority. Higher-priority Pods are considered for preemption only if the removal of the lowest priority Pods is not sufficient to allow the scheduler to schedule the preemptor Pod, or if the lowest priority Pods are protected by PodDisruptionBudget
[source](https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/#interactions-of-pod-priority-and-qos)

This makes me think that the normal job pods are evicted first, as they have the lowest priority. Why don't we just change the Preemption like [here](https://github.com/kubevirt/project-infra/pull/2068/files#diff-56380dd553d457f6b73e214c4f728aa374364c300c133bbceddac81d8aa3f0f4R7) and preempt the job pods (which will be restarted by prow-controller-manager anyway)?

Reply:
I was thinking that the regular pods with the lower priority belong to some default priority class of k8s which is not preemptive, @oshoval please keep me honest here. I will double check it

Reply:
@dhiller 
Since the DS has the label that the sriov has anti-affinity to
https://github.com/kubevirt/project-infra/pull/2068/files#diff-894e55e0973533b54e923ff56852bf9a6ec68892d7c9a97cc1a251a370eddb80R17
the place-holder pod will be evicted first, even that the default pods have lower priority
We don't want the sriov to stop random running jobs, but specifically the place-holder, there fore the following design.

@enp0s3 the default pods don't have pc, because there isn't a global default pc, so they have priority zero,
it is not preemptive, which means they can't preempt. but they still can be preempted,
but because the above reason (label and anti-affinity), the place-holder will be the one that is preempted